demo/clusters/openshift/README.md

```diff
@@ -0,0 +1,142 @@
+# Running the NVIDIA DRA Driver on Red Hat OpenShift
+
+This document explains the differences between deploying the NVIDIA DRA driver on OpenShift and upstream Kubernetes or its flavors.
+
+## Prerequisites
+
+Install a recent build of OpenShift 4.16 (e.g. 4.16.0-ec.3). You can obtain an IPI installer binary (`openshift-install`) from the [Release Status](https://amd64.ocp.releases.ci.openshift.org/) page, or use the Assisted Installer to install on bare metal. Refer to the [OpenShift documentation](https://docs.openshift.com/container-platform/4.15/installing/index.html) for different installation methods.
+
+## Enabling DRA on OpenShift
+
+Enable the `TechPreviewNoUpgrade` feature set as explained in [Enabling features using FeatureGates](https://docs.openshift.com/container-platform/4.15/nodes/clusters/nodes-cluster-enabling-features.html), either during the installation or post-install. The feature set includes the `DynamicResourceAllocation` feature gate.
+
+Update the cluster scheduler to enable the DRA scheduling plugin:
+
+```console
+$ oc patch --type merge -p '{"spec":{"profile": "HighNodeUtilization", "profileCustomizations": {"dynamicResourceAllocation": "Enabled"}}}' scheduler cluster
+```
+
+## NVIDIA GPU Drivers
+
+The easiest way to install NVIDIA GPU drivers on OpenShift nodes is via the NVIDIA GPU Operator.
+
+**Be careful to disable the device plugin so it does not conflict with the DRA plugin**. It is recommended to leave only the NVIDIA GPU driver and driver toolkit configs, and disable everything else:
+
+```yaml
+  <...>
+  devicePlugin:
+    enabled: false
+  <...>
+  driver:
+    enabled: true
+  <...>
+  toolkit:
+    enabled: true
+  <...>
```

Comment:
@elezar I'm not sure disabling everything except for the driver and the toolkit is such a good idea. I will double-check.

Reply:
The `devicePlugin` should definitely be disabled on nodes where the DRA driver is to be used. The other component that we would need to check is the MIG manager. This *could* be used to set the mig mode on the devices, but SHOULD not be used to actually create MIG profiles.

Reply:
@elezar it looks like when DCGM/DCGM exporter are enabled in the GPU operator, trying to enable MIG will result in 
```console
Warning: MIG mode is in pending enable state for GPU 00000001:00:00.0:In use by another client
00000001:00:00.0 is currently being used by one or more other processes (e.g. CUDA application or a monitoring application such as another instance of nvidia-smi). Please first kill all processes using the device and retry the command or reboot the system to make MIG mode effective.
```
I don't think they're needed for DRA and can be safely disabled. I will mention it in the doc.

Reply:
DCGM should be shutdown and restarted by the MIG manager as seen here:
https://github.com/NVIDIA/mig-parted/blob/main/deployments/gpu-operator/reconfigure-mig.sh#L436

Is this failing in your environment for some reason?